gae-to-gae-auth/client/client.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import json
+import logging
+import os
+import requests
+import sys
+import time
+
+from google.auth import crypt, jwt
+
+GOOGLE_ID_AUDIENCE = "https://www.googleapis.com/oauth2/v4/token"
+
+def make_signed_jwt(service_account_key_path, audience, subject=None, ttl_seconds=3600, **kwargs):
```

Comment:
After spending quite a bit of time learning about auth, I believe we should (eventually) be able to use https://github.com/google/oauth2client, although I suspect there are a few bugs that currently prevents us from using it correctly.

Reply:
Well, it turns out oauth2client is deprecated and to be replaced by https://github.com/GoogleCloudPlatform/google-auth-library-python
I'll need to look at this one and see if we can use it instead...